gobblin-runtime/src/main/java/gobblin/runtime/Fork.java

```diff
@@ -125,23 +216,70 @@ public void updateByteMetrics()
   /**
    * Commit data of this {@link Fork}.
    *
-   * @param recordsPulled number of records pulled
-   * @param expectedCount expected record count
-   * @param pullLimit pull limit
-   * @throws Exception
+   * @throws Exception if there is anything wrong committing the data
    */
-  public void commit(long recordsPulled, long expectedCount, long pullLimit)
+  public void commit()
       throws Exception {
-    if (checkDataQuality(recordsPulled, expectedCount, pullLimit, this.convertedSchema)) {
-      // Commit data if all quality checkers pass. Again, not to catch the exception
-      // it may throw so the exception gets propagated to the caller of this method.
-      commitData();
+    try {
+      if (checkDataQuality(this.convertedSchema)) {
+        // Commit data if all quality checkers pass. Again, not to catch the exception
+        // it may throw so the exception gets propagated to the caller of this method.
+        this.logger.info(String.format("Committing data for fork %d of task %s", this.index, this.taskId));
+        commitData();
+        setForkState(ForkState.SUCCEEDED, ForkState.COMMITTED);
```

Comment:
What does it mean if we have committed the data (commitData()) but setForkState() fails? Should we just set the state explicitly to COMMITTED.